src/test/java/gregtech/common/metatileentities/storage/QuantumChestTest.java

```diff
@@ -0,0 +1,248 @@
+import static gregtech.api.util.GTTransferUtils.insertItem;
+import static gregtech.api.util.GTUtility.gregtechId;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class QuantumChestTest {
+    private static ItemStack GRAVEL;
+    private static ItemStack SAND;
+
+    @BeforeAll
+    public static void bootstrap() {
+        Bootstrap.perform();
+        GRAVEL = new ItemStack(Blocks.GRAVEL);
+        GRAVEL.setCount(64);
+        SAND = new ItemStack(Blocks.SAND);
+        SAND.setCount(64);
+    }
+
+    @Test
+    public void Test_Valid() {
+        for (var quantumChest : createInstances()) {
+            assertThat(quantumChest, notNullValue());
+        }
+    }
+
+    @Test
+    public void Test_Single_Insertion() {
+        for (var quantumChest : createInstances()) {
+            IItemHandler combinedInventory = quantumChest.getCombinedInventory();
+            IItemHandler virtualInventory = quantumChest.getItemInventory();
+            IItemHandlerModifiable exportItems = quantumChest.getExportItems();
+            IItemHandlerModifiable importItems = quantumChest.getImportItems();
+
+            ItemStack stack = insertItem(combinedInventory, GRAVEL.copy(), true);
+            String reason = String.format("%s should be Empty!", itemStackToString(stack));
+            assertThat(reason, stack.isEmpty());
+
+            insertItem(combinedInventory, GRAVEL.copy(), false);
+            int expected = 64;
+            stack = exportItems.getStackInSlot(0);
+            reason = String.format("Got %d in the export slot when it should be %d", exportItems.getStackInSlot(0).getCount(), expected);
+            assertThat(reason, stack.getCount() == expected);
+
+            stack = insertItem(combinedInventory, GRAVEL.copy(), true);
+            reason = String.format("%s should be Empty!", itemStackToString(stack));
+            assertThat(reason, stack.isEmpty());
+
+            insertItem(combinedInventory, GRAVEL.copy(), false);
+            stack = virtualInventory.getStackInSlot(0);
+            reason = String.format("Got %d in the virtualized inventory when it should be %d", stack.getCount(), expected);
+            assertThat(reason, stack.getCount() == expected);
+
+            stack = importItems.insertItem(0, SAND.copy(), true);
+            reason = String.format("%s should not be Empty!", itemStackToString(stack));
+            assertThat(reason, !stack.isEmpty());
+
+            stack = importItems.insertItem(0, SAND.copy(), false);
+            reason = String.format("%s should be Sand!", itemStackToString(stack));
+            assertThat(reason, !stack.isEmpty());
+
+            stack = virtualInventory.getStackInSlot(0);
+            reason = String.format("Got %d in the virtualized inventory when it should be %d", stack.getCount(), expected);
+            assertThat(reason, stack.getCount() == expected);
```

Comment:
Try using the `is()` matcher for comparing the remainder and stack counts for all of these tests. This will allow both values to be visible in the log in case the test fails.